src/unix/index_unix.ml

```diff
@@ -82,13 +109,15 @@ module IO : Index.IO = struct
       really_write t.fd off buf;
       t.cursor <- off ++ Int64.of_int (Bytes.length buf);
       stats.bytes_written <- stats.bytes_written + Bytes.length buf;
-      stats.nb_writes <- succ stats.nb_writes
+      stats.nb_writes <- succ stats.nb_writes;
+      Metrics.add src no_tags (fun m -> m stats)
```

Comment:
I don't think we should mix `Metrics` and `stats` updates -- it's a complex subject but I think a good architecture would be to:
- expose more information in `stats` as you did
- add metrics and reporters at the call site (e.g. in the benchmarks?) where metrics API calls are done with the contents of `stats`.

It will be a bit less precise, but we can control the amount of things reported more easily and it will not mix various concerns in the core library. What do you think?

Reply:
Then we can't tell how many bytes are read/ written per io read /write, but maybe it's ok.

Reply:
Yes I agree that we might loose some accuracy, but in that case the final caller can do `bytes_written / nb_writes` to get a reasonable average estimate. But you are right that it might be interesting o see a proper distribution instead of an average... I think that can be done later.